master/internal/rm/agentrm/provisioner/provisioner.go

```diff
@@ -164,3 +184,19 @@ func (p *Provisioner) provision(ctx *actor.Context) {
 			p.InstanceType())
 	}
 }
+
+func (p *Provisioner) launch(ctx *actor.Context, numToLaunch int) error {
+	if err := p.launchErr.GetError(); err != nil {
+		return err
+	}
+	p.launchErr.SetError(p.provider.launch(ctx, numToLaunch))
+	return p.launchErr.GetError()
+}
+
+// GetLaunchError returns the current launch error sent from the provider.
+func (p *Provisioner) GetLaunchError() error {
+	if p == nil {
+		return nil
+	}
```

Comment:
ah.. i see this is why we weren't nil checking the provisioner. i dont really like this though, i think it is clearer to nil check the provisioner in the RP code and not call it when we dont have a provisioner.

Reply:
updated, unfortunately it just adds more checks in other places